test/externalTests/prb-math.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import sys
+import subprocess
+import json
+from pathlib import Path
+from shutil import copyfile
+
+from external_test import TestConfig, ExternalTest, CURRENT_EVM_VERSION
+from foundry import FoundryRunner
+
+class PRBMath(ExternalTest):
+    """PRBMath test setup"""
+
+    def __init__(self, config: TestConfig):
+        ExternalTest.__init__(self, config)
+
+    def setup_fn(self, test_dir: Path, env: dict):
+        """ PRBMath setup steps"""
+        copyfile(test_dir / ".env.example", test_dir / ".env")
+
+    def compile_fn(self, test_dir: Path, env: dict):
+        """ PRBMath compilation steps"""
+        # TODO
+
+    def test_fn(self, test_dir: Path, env: dict):
+        """ PRBMath tests steps"""
+        # TODO
+
+
+if __name__ == '__main__':
+    try:
+        # Note: We currently accept solc_binary_type and solc_binary_path as parameters
+        # to keep compatibility with the current external tests framework.
+        args = ExternalTest.parse_command_line(
+            "PRBMath external tests", sys.argv[1:])
+
+        config_json = json.loads(f"""
```

Comment:
Wouldn't it make more sense for `TestConfig` to have a builder or something like, and then if need be dump to JSON. Having to write out a JSON object by hand seems a bit much.

Reply:
I'd say that even without a builder this is completely unnecessary. You could use proper Python dicts here and you'd get all the normal code checks as a bonus. And the syntax is very close to JSON anyway.